packages/api-plugin-fulfillment/src/simpleSchemas.js

```diff
@@ -0,0 +1,169 @@
+import SimpleSchema from "simpl-schema";
+
+/**
+ * @name MethodEmptyData
+ * @memberof Schemas
+ * @type {SimpleSchema}
+ * @summary Defines Empty placeholder
+ * @property {String} gqlType Defines the method type
+ * @property {Boolean} emptyData Empty Data fields
+ */
+export const MethodEmptyData = new SimpleSchema({
+  gqlType: String,
+  emptyData: {
+    type: Boolean,
+    optional: true
+  }
+});
+
+/**
+ * @name FulfillmentMethodSchema
+ * @memberof Schemas
+ * @type {SimpleSchema}
+ * @summary Defines Schema for Fulfillment Method
+ * @property {Number} cost - optional cost
+ * @property {String[]} fulfillmentTypes - fulfillment type retained for backward compatibility
+ * @property {String} group - Group to which fulfillment method belong
+ * @property {Number} handling - handling charges
+ * @property {Boolean} enabled - status of fulfillment method
+ * @property {String} label - displayed on the UI
+ * @property {String} name - name of the fulfillment method, user editable
+ * @property {String} fulfillmentMethod - used by application, not user editable
+ * @property {String} displayMessageMethod - used to display additional message on UI specific to ff-method
+ * @property {Number} rate - ratefor themethod
+ */
+export const FulfillmentMethodSchema = new SimpleSchema({
+  "cost": {
+    type: Number,
+    optional: true
+  },
+  "fulfillmentTypes": {
+    type: Array,
+    minCount: 1
+  },
+  "fulfillmentTypes.$": String,
+  "group": String,
+  "handling": Number,
+  "enabled": Boolean,
+  "label": String,
+  "name": String,
+  "fulfillmentMethod": {
+    type: String,
+    optional: true
+  },
+  "displayMessageMethod": {
+    type: String,
+    optional: true
+  },
+  "rate": Number
+});
+
+/**
+ * @name ProviderSchema
+ * @memberof Schemas
+ * @type {SimpleSchema}
+ * @summary Defines Schema for Provider details inside Fulfillment Type
+ * @property {Boolean} enabled - status of fulfillment method
+ * @property {String} label - displayed on the UI
+ * @property {String} name - name of the fulfillment type, user editable
+ */
+const ProviderSchema = new SimpleSchema({
+  enabled: Boolean,
+  label: String,
+  name: String
+});
+
+/**
+ * @name FulfillmentTypeSchema
+ * @memberof Schemas
+ * @type {SimpleSchema}
+ * @summary Defines Schema for Fulfillment Type
+ * @property {String} name - name of the fulfillment type, user editable
+ * @property {String} shopId - Shop ID
+ * @property {ProviderSchema} provider - Provider details
+ * @property {String} fulfillmentType - fulfillmentType, not user editable
+ * @property {String} displayMessageType - used to display additional message on UI specific to ff-type
+ * @property {FulfillmentMethodSchema[]} methods - array of ff-methods under this ff-type
+ */
+export const FulfillmentTypeSchema = new SimpleSchema({
+  "name": String,
+  "shopId": String,
+  "provider": {
+    type: ProviderSchema
+  },
+  "fulfillmentType": String,
```

Comment:
Maybe something like `baseType` would be clearer than `fulfillmentType` since you're already calling each document in this collection a "fulfillment type"?

Reply:
Hi @aldeed 
Generally I think the field name `baseType` inside the collection should be fine.

But considering that we have the collection as `Fulfillment` with the 2 major components as `fulfillmentType` and `fulfillmentMethod` with a logical association (in the names), I feel like `baseType` could create a slight confusion.

Also, this fix would require changes in many files across multiple plugins which increases the probability of breaking something.

This is just what I thought about it (not a very strong opinion) and please let me know if you feel strongly about `baseType` and I can make the changes